File: lista03W/zad2/prostokat.cpp
#include "prostokat.hpp"

Prostokat::Prostokat(double b1, double b2) : Czworokat(b1, b2, b1, b2, 90) {
  if(b1 <= 0 || b2 <= 0)
    throw std::invalid_argument("Boki prostokata musza byc dodatnie.");
}

double Prostokat::getPole() const {
  return bok1 * bok2;
}

double Prostokat::getObwod() const {
  return 2 * (bok1 + bok2);
}

std::string Prostokat::getNazwa() const {
  return "Prostokat";
}

File: lista03W/zad1/Prostokat.java
public class Prostokat extends Czworokat {
  public Prostokat(double bok1, double bok2) {
    super(bok1, bok2, bok1, bok2, 90);
  }
  
  public double getPole() {
    return bok1 * bok2;
  }
  
  public double getObwod() {
    return 2 * (bok1 + bok2);
  }
  
  public String getNazwa() {
    return "Prostokat";
  }
}